### ISSUE TITLE
Dropbear fails to start - XCRYPT_2.0 not found

### ISSUE THREAD

**Issue by oxedions** [ivandavidov/minimal#220]
Hello,

First, thanks a lot for MLL. It is a very nice tool 🙂 

I have an issue with latest version and dropbear.

When starting dropbear, I have error:

```
/ # dropbear
dropbear: /lib/libcrypt.so.&: version 'XCRYPT_2.0' not found (required by dropbear)
```

I found on the web that this issue seems common recently (2019-2020).

I tried to upgrade to https://matt.ucc.asn.au/dropbear/releases/dropbear-2020.80.tar.bz2 but this does not seems to be related to dropbear at all, but more to a change in the libcrypt/glibc.

* https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1867423
* https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1867432
* https://patchwork.ozlabs.org/project/glibc/patch/20170829184247.6875-1-zackw@panix.com/

By the way, I can now start MLL by PXE. If you desire it, I can add a small note in the README (PR) describing how to simply achieve that.

With my best regards

Ox

Notes on building system:
* Centos 8
* gcc (GCC) 8.3.1 20191121 (Red Hat 8.3.1-5)
```# rpm -qa | grep glibc
glibc-devel-2.28-101.el8.x86_64
glibc-2.28-101.el8.x86_64
glibc-headers-2.28-101.el8.x86_64
glibc-common-2.28-101.el8.x86_64
glibc-langpack-en-2.28-101.el8.x86_64
```

**Comment by ivandavidov** [comment]
Thanks for bringing this up. I'll take a look at Dropbear as soon as I have the time for that.

Meanwhile, feel free to send a PR regarding the PXE boot sequence. Improvements are always welcome.

**Comment by oxedions** [comment]
Thanks a lot 😊 

PR done https://github.com/ivandavidov/minimal/pull/221

**Comment by ivandavidov** [comment]
Thank you for the quality work. I merged your PR.

**Comment by oxedions** [comment]
Thank you ! I made a small mistake, so created a second PR. Sorry for that 🙁 

I tried to include **stress-ng** into bundles (advanced **stress** version), as I need it, but I also end up in the same `version 'XCRYPT_2.0' not found` than for dropbear. It seems to impact all that rely on libcrypt.

**Comment by ivandavidov** [comment]
The solution to the problem is to build ``xcrypt`` which provides the correct version of ``libcrypt.so.1``.

https://github.com/besser82/libxcrypt

I already tested this and dropbear works fine. The overall steps are the following:

1. Download xcrypt
2. Configure xcrypt with ``--prefix`` somewhere
3. make + make install = libxcrypt with correct version
3. Overwrite the original `MLL `libcrypt.so.1`` library with the one provided by xcrypt

From MLL point of view all we need is to provide ``xcrypt`` as additional overlay bundle and the issue will be resolved automagically.

**Comment by ivandavidov** [comment]
More background to the issue.

All compile-time library dependencies for pretty much all overlay bundles are resolved against the host machine. Most notably, in MLL we specify our own GLIBC version which almost certainly is different from the GLIBC version on the host machine.

When we build e.g. Dropbear, we compile it against the host machine but at runtime the shared objects come from the MLL build process which may cause behavioral inconsistencies.

In this particular case the official ``crypt`` library is _kind-of-deprecated_ and it no longer comes from GLIBC (at least not the version that we need). It comes from the ``xcrypt`` project which provides compatible shared object library.

There are two possible solutions.

1. Provided the correct version of the ``crypt`` library at runtime. This is the option that I explained in my previous post.
2. Link all overlay bundles against the GLIBC version which we have built as part of the MLL's build process.

Architecturally speaking the second option is better, yet much harder to achieve, because there is no simple and elegant way to just "tell" the overlay bundles to use different sysroot. This must be configured individually for each overlay bundle.
